client-runtime/src/main/java/com/microsoft/rest/v2/RestProxy.java

```diff
@@ -46,58 +32,45 @@
-        final String methodName = method.getName();
-        final SwaggerMethodProxyDetails methodDetails = interfaceDetails.getMethodProxyDetails(methodName);
-
-        final String actualHost = methodDetails.applyHostSubstitutions(host, args);
-        final String[] hostParts = actualHost.split("://");
-        final String actualPath = methodDetails.getSubstitutedPath(args);
+        final SwaggerMethodParser methodParser = interfaceParser.methodParser(method);
 
         final UrlBuilder urlBuilder = new UrlBuilder()
-                .withScheme(hostParts[0])
-                .withHost(hostParts[1])
-                .withPath(actualPath);
+                .withScheme(methodParser.scheme(args))
+                .withHost(methodParser.host(args))
+                .withPath(methodParser.path(args));
 
-        for (EncodedParameter queryParameter : methodDetails.getEncodedQueryParameters(args)) {
+        for (final EncodedParameter queryParameter : methodParser.encodedQueryParameters(args)) {
             urlBuilder.withQueryParameter(queryParameter.name(), queryParameter.encodedValue());
         }
 
         final String url = urlBuilder.toString();
-        final HttpRequest request = new HttpRequest(methodDetails.method(), url);
-
-        for (final EncodedParameter headerParameter : methodDetails.getEncodedHeaderParameters(args)) {
-            request.withHeader(headerParameter.name(), headerParameter.encodedValue());
-        }
+        final HttpRequest request = new HttpRequest(methodParser.fullyQualifiedMethodName(), methodParser.httpMethod(), url);
 
-        for (final HttpHeader header : methodDetails.getHeaders()) {
-            request.withHeader(header.getName(), header.getValue());
+        for (final HttpHeader header : methodParser.headers(args)) {
+            request.withHeader(header.name(), header.value());
         }
 
-        final Integer bodyContentMethodParameterIndex = methodDetails.bodyContentMethodParameterIndex();
-        if (bodyContentMethodParameterIndex != null) {
-            final Object bodyContentObject = args[bodyContentMethodParameterIndex];
-            if (bodyContentObject != null) {
-                final String bodyContentString = serializer.serialize(bodyContentObject);
-                request.withBody(bodyContentString, "application/json");
-            }
+        final Object bodyContentObject = methodParser.body(args);
+        if (bodyContentObject != null) {
+            final String mimeType = "application/json";
+            final String bodyContentString = serializer.serialize(bodyContentObject);
+            request.withBody(bodyContentString, mimeType);
         }
 
-        Object result = null;
-        final Type returnType = methodDetails.returnType();
+        Object result;
+        final Type returnType = methodParser.returnType();
         final TypeToken returnTypeToken = TypeToken.of(returnType);
         if (returnTypeToken.isSubtypeOf(Completable.class)) {
```

Comment:
I spent some time trying to think about how to properly parse an Observable<> return type, but I just couldn't think of the right way to do it. I think that was because I'm not sure there's an appropriate use case. Paging may be a good example, but I need to see an example of a paging request before I code that up.
Since that will be an easy thing to add later, I'm fine with postponing it until then.